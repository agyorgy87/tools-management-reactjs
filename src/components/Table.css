@import url('https://fonts.googleapis.com/css2?family=Jura&display=swap');

.theTable {
    width: 100%;
    margin-top: 20px;
    border-spacing: 20px;
}

tr {
    height: 100px; 
    color: white;
    font-family: 'Jura', sans-serif;
}

th {
    color: #fe8e22;
    font-size: 180%;
    font-family: 'Jura', sans-serif;
}

#name {
    width:20%;
}

#description {
    width: 20%;
}

#price {
    width: 20%;
}

#quantity {
    width: 20%;
}