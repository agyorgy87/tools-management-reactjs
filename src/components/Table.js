import '../css/Table.css';
import React from 'react';

export default function Table (props) {
    
    return (
        <div className="table-container"> 
            <table>
                <thead>
                    <tr className="mainBoardTitle">
                        <th id="name">Name</th>
                        <th id="description">Explanation</th>
                        <th id="quantity">Stock Of Store</th>
                        <th id="price">Price</th>
                    </tr>
                </thead>
                <tbody>
                    {
                        props.data.map((row, index) => (
                            <tr key={"tools" + index}>
                                <td>{row.Name}</td>
                                <td>{row.Description}</td>
                                <td>{row.Quantity}</td>
                                <td>{row.Price}</td>
                            </tr>
                        ))
                    }
                </tbody>
            </table>
        </div>
    )
}