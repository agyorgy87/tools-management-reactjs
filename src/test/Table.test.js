import renderer from 'react-test-renderer';
import Table from "../components/Table.js";

test("test empty array", () => {
    let tree = renderer
    .create(<Table data={[]}/>)
    .toJSON()
    expect(tree).toMatchSnapshot();
});

test("test with data", () => {
    let list = [{Name: "Feri", Description: "well done", Quantity: "1", Price: "1000"}];
    let tree = renderer
    .create(<Table data={list}/>)
    .toJSON()
    expect(tree).toMatchSnapshot();
});