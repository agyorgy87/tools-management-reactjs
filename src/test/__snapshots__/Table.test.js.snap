// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test empty array 1`] = `
<div
  className="table-container"
>
  <table>
    <thead>
      <tr
        className="mainBoardTitle"
      >
        <th
          id="name"
        >
          Name
        </th>
        <th
          id="description"
        >
          Explanation
        </th>
        <th
          id="quantity"
        >
          Stock Of Store
        </th>
        <th
          id="price"
        >
          Price
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`test with data 1`] = `
<div
  className="table-container"
>
  <table>
    <thead>
      <tr
        className="mainBoardTitle"
      >
        <th
          id="name"
        >
          Name
        </th>
        <th
          id="description"
        >
          Explanation
        </th>
        <th
          id="quantity"
        >
          Stock Of Store
        </th>
        <th
          id="price"
        >
          Price
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Feri
        </td>
        <td>
          well done
        </td>
        <td>
          1
        </td>
        <td>
          1000
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
